system/include/compat/emmintrin.h

```diff
@@ -1386,16 +1386,7 @@ _mm_packus_epi16(__m128i __a, __m128i __b)
 static __inline__ int __attribute__((__always_inline__, __nodebug__))
 _mm_movemask_epi8(__m128i __a)
 {
-  // TODO: optimize
-  union {
-    unsigned char x[16];
-    __m128i m;
-  } src;
-  src.m = __a;
-  unsigned int x = 0;
-  for(int i = 0; i < 16; ++i)
-    x |= ((unsigned int)src.x[i] >> 7) << i;
-  return (int)x;
+  return (int)wasm_i8x16_bitmask((v128_t)__a);
```

Comment:
```suggestion
  return (int)wasm_i8x16_bitmask(__a);
```

Most of the other functions do not include casts to `v128_t`.

Reply:
Done

Reply:
Doesn't compile without casts, reverting back

Reply:
Ah, the __m128i doesn't need the casts, but __m128d still does. Anyway, it's probably fine to keep all the casts.